lib/puppet/configurer.rb

```diff
@@ -528,6 +533,14 @@ def retrieve_new_catalog(query_options)
   end
 
   def download_plugins(remote_environment_for_plugins)
-    @handler.download_plugins(remote_environment_for_plugins)
+    begin
+      @handler.download_plugins(remote_environment_for_plugins)
+    rescue Puppet::Error => detail
+      if !Puppet[:ignore_plugin_errors] && Puppet[:usecacheonfailure]
+        @pluginsync_failed = true
+      else
+        raise detail
```

Comment:
Should we raise a `Puppet::Error` here? Similar to https://github.com/puppetlabs/puppet/blob/e31583dcd9ec1f897ae1c58079fc3c108661c77b/lib/puppet/configurer.rb#L54

Reply:
The exception was probably raised here: 

https://github.com/ciprianbadescu/puppet/blob/cf2f609c1b470a9a0e285ae3c33181d216d044b1/lib/puppet/configurer/downloader.rb#L21

And logged here if ignored

https://github.com/ciprianbadescu/puppet/blob/cf2f609c1b470a9a0e285ae3c33181d216d044b1/lib/puppet/configurer/downloader.rb#L32

The new code rescue error only if `!Puppet[:ignore_plugin_errors] && Puppet[:usecacheonfailure]`, otherwise is raised like it was before (`raise detail`).